merlin/models/tf/loader.py

```diff
@@ -289,288 +276,78 @@ def __init__(
         )
         if schema:
             dataset.schema = schema
+
+        cat_names = cat_names or (
+            dataset.schema.select_by_tag(Tags.CATEGORICAL)
+            .excluding_by_tag(Tags.TARGET)
+            .column_names
+            if _get_schema(dataset)
+            else []
+        )
+        cont_names = cont_names or (
+            dataset.schema.select_by_tag(Tags.CONTINUOUS).excluding_by_tag(Tags.TARGET).column_names
+            if _get_schema(dataset)
+            else []
+        )
+        label_names = label_names or (
+            dataset.schema.select_by_tag(Tags.TARGET).column_names if _get_schema(dataset) else []
+        )
+
         cat_names, cont_names, label_names = _validate_schema(
             feature_columns, cat_names, cont_names, label_names, schema=dataset.schema
         )
 
+        dataset.schema = _augment_schema(
```

Comment:
What is the use-case for this `_augment_schema` ? Is there something that depends on this functionality?